state-chain/pallets/cf-elections/src/lib.rs

```diff
@@ -1513,14 +1541,28 @@ pub mod pallet {
 		}
 
 		#[pallet::call_index(36)]
-		#[pallet::weight(Weight::zero())] // TODO: Benchmarks
+		#[pallet::weight(T::WeightInfo::clear_all_votes(
+			WORST_CASE_CLEAR_ALL_VOTE_STORAGE_LEN,
+			WORST_CASE_CLEAR_ALL_VOTE_STORAGE_LEN,
+			WORST_CASE_CLEAR_ALL_VOTE_STORAGE_LEN,
+			WORST_CASE_CLEAR_ALL_VOTE_STORAGE_LEN,
+			WORST_CASE_CLEAR_ALL_VOTE_STORAGE_LEN
+		))]
 		pub fn clear_all_votes(
 			origin: OriginFor<T>,
 			limit: u32,
 			ignore_corrupt_storage: CorruptStorageAdherance,
-		) -> DispatchResult {
+		) -> DispatchResultWithPostInfo {
 			Self::ensure_governance(origin, ignore_corrupt_storage)?;
 
+			let weight = T::WeightInfo::clear_all_votes(
+				SharedDataReferenceCount::<T, I>::iter_keys().count() as u32,
+				SharedData::<T, I>::iter_keys().count() as u32,
+				BitmapComponents::<T, I>::iter_keys().count() as u32,
+				IndividualComponents::<T, I>::iter_keys().count() as u32,
+				ElectionConsensusHistoryUpToDate::<T, I>::iter_keys().count() as u32,
```

Comment:
Oh i was just thinking - in case when we want to clear everything and we just use u32::MAX as the "limit". 
but I can see this is a Governance extrinsic so we will probably put in sensible numbers. right?